apps/modernization-api/src/main/java/gov/cdc/nbs/patient/delete/PatientDeleteController.java

```diff
@@ -1,23 +1,56 @@
 package gov.cdc.nbs.patient.delete;
 
-import gov.cdc.nbs.model.PatientEventResponse;
+import gov.cdc.nbs.authentication.NbsUserDetails;
+import gov.cdc.nbs.config.security.SecurityUtil;
+import gov.cdc.nbs.patient.PatientException;
+import gov.cdc.nbs.patient.RequestContext;
+import gov.cdc.nbs.patient.search.PatientSearchIndexer;
 import org.springframework.graphql.data.method.annotation.Argument;
 import org.springframework.graphql.data.method.annotation.MutationMapping;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.stereotype.Controller;
 
+import java.time.Clock;
+import java.time.Instant;
+
 @Controller
 class PatientDeleteController {
 
-    private final PatientDeletionRequester requester;
+    private final Clock clock;
+    private final PatientDeleter deleter;
+    private final PatientSearchIndexer indexer;
 
-    PatientDeleteController(final PatientDeletionRequester requester) {
-        this.requester = requester;
+    PatientDeleteController(
+        final Clock clock,
+        final PatientDeleter deleter,
+        final PatientSearchIndexer indexer
+    ) {
+        this.clock = clock;
+        this.deleter = deleter;
+        this.indexer = indexer;
     }
 
     @MutationMapping("deletePatient")
     @PreAuthorize("hasAuthority('VIEW-PATIENT') and hasAuthority('DELETE-PATIENT')")
-    public PatientEventResponse delete(@Argument("patientId") long patient) {
-        return requester.request(patient);
+    PatientDeleteResult delete(@Argument long patient) {
+
+        NbsUserDetails user = SecurityUtil.getUserDetails();
+
+        RequestContext context = new RequestContext(user.getId(), Instant.now(this.clock));
+
+        try {
+
+            deleter.delete(context, patient);
+            this.indexer.index(patient);
```

Comment:
is this deleting the document in elastic or reindexing it?

Reply:
It's re-indexing it, the status changes to `LOG_DEL` when a patient is deleted.  There is a use case where certain users can search over deleted patients

Reply:
Cool